### PR TITLE
Add FreeBSD to CI

### PR DESCRIPTION
**Pull request by jbeich**
TravisCI [supports](https://github.com/travis-ci/travis-build/pulls?q=is%3Apr+freebsd+is%3Aclosed) FreeBSD nowadays. Let's see how well it works.

- Only libusb is supported, so it won't regress anymore
- Deployment is not implemented because why bother given downstream package

### CODE REVIEW

**Comment by whoozle** [comment]
is it possible to add `readline` library to this PR?

**Comment by jbeich** [comment]
readline seems like a bug, investigating. I also need to add pybind11 dependency.

**Comment by jbeich** [comment]
Oops, I haven't fixed readline bug completely.

File: .travis.yml [hunk 1]
```diff
@@ -32,6 +36,9 @@ script:
       export VERSION=$(git rev-parse --short HEAD) &&
       ./linuxdeployqt-continuous-x86_64.AppImage appdir/usr/share/applications/*.desktop -bundle-non-qt-libs &&
       ./linuxdeployqt-continuous-x86_64.AppImage appdir/usr/share/applications/*.desktop -appimage
+    elif [ "${TRAVIS_OS_NAME}" = "freebsd" ]; then
+      cmake -B build -G Ninja -DCMAKE_BUILD_TYPE=Release -DUSB_BACKEND_LIBUSB=YES
```
Comment (jbeich) [inline .travis.yml:40]: `-DCMAKE_BUILD_TYPE=Release` looks unnecessary given
https://github.com/whoozle/android-file-transfer-linux/blob/2a86c5956150d9d973514f71aec559038ec436ad/CMakeLists.txt#L89-L91

File: cli/CMakeLists.txt [hunk 2]
```diff
@@ -9,6 +9,7 @@ set(CLI_SOURCES
 	cli.cpp)
 
 if (READLINE_FOUND)
+	set(CMAKE_REQUIRED_INCLUDES ${Readline_INCLUDE_DIR})
```
Comment (jbeich) [inline cli/CMakeLists.txt:12]: Oops, I forgot to clear `CMAKE_REQUIRED_INCLUDES` after the check.

File: .travis.yml [hunk 3]
```diff
@@ -26,7 +26,7 @@ install:
 script:
   - |
     if [ "${TRAVIS_OS_NAME}" = "linux" ]; then
-      cmake . -DCMAKE_BUILD_TYPE=Release -DCMAKE_INSTALL_PREFIX=/usr &&
```
Comment (whoozle) [inline .travis.yml:25]: please undo this, that if() in cmake does not reliably work.

File: cli/CMakeLists.txt [hunk 4]
```diff
@@ -35,12 +36,14 @@ int main(int argc, char **argv) {
...
 	endif()
 
+	set(CMAKE_REQUIRED_INCLUDES)
 	set(CMAKE_REQUIRED_LIBRARIES)
 endif()
 
 if (READLINE_FOUND)
 	list(APPEND CLI_SOURCES CommandLine.cpp)
 	list(APPEND CLI_LIBRARIES ${Readline_LIBRARY})
+	include_directories(SYSTEM ${Readline_INCLUDE_DIR})
```
Comment (whoozle) [inline cli/CMakeLists.txt:46]: please use `target_include_directories` for cli target (I reckon it should be PRIVATE)

**Comment by jbeich** [comment]
Anything else? I've made changes as you've suggested but not sure if you've got notification from force push and resolving review comments.

**Comment by whoozle** [comment]
Thank you for your work! 

If you want more work on AFTL, you can try implementing direct FreeBSD backend without libusb. Libusb has serious limitations, at least at the time I started this project.

**Comment by jbeich** [comment]
FreeBSD and DragonFly don't use the reference libusb implemantiton, so Linux libusb bugs may not translate. ;)

**Comment by whoozle** [comment]
there's API limitation, not only bugs, e.g. you have to provide full data for bulk transfer in memory,  limitations in description parsing, etc etc

**Pull request merged**